shopfloor/actions/stock.py

```diff
@@ -0,0 +1,252 @@
+# Copyright 2020 Camptocamp SA (http://www.camptocamp.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import _, fields
+from odoo.tools.float_utils import float_round
+
+from odoo.addons.component.core import Component
+
+from ..exceptions import ConcurentWorkOnTransfer
+
+
+class StockAction(Component):
+    """Provide methods to work with stock operations."""
+
+    _name = "shopfloor.stock.action"
+    _inherit = "shopfloor.process.action"
+    _usage = "stock"
+
+    def _create_return_move__get_max_qty(self, origin_move):
+        """Returns the max returneable qty."""
+        # The max returnable qty is the sent qty minus the already returned qties
+        quantity = origin_move.product_uom_qty
+        for move in origin_move.move_dest_ids:
+            if (
+                move.origin_returned_move_id
+                and move.origin_returned_move_id != origin_move
+            ):
+                continue
+            if move.state in ("partially_available", "assigned"):
+                quantity -= sum(move.move_line_ids.mapped("quantity_product_uom"))
+            elif move.state in ("done"):
+                quantity -= move.product_uom_qty
```

Comment:
```suggestion
                quantity -= move.quantity
```
I guess the same field can be used for the condition above